daemon/response.go

```diff
@@ -102,56 +135,29 @@ func maintenanceForRestartType(rst state.RestartType) *errorResult {
 	return e
 }
 
-func (r *resp) addWarningsToMeta(count int, stamp time.Time) {
-	if r.Meta != nil && r.Meta.WarningCount != 0 {
+func (r *respJSON) addMaintenanceFromRestartType(rst state.RestartType) {
+	if rst == state.RestartUnset {
+		// nothing to do
 		return
 	}
-	if count == 0 {
+	r.Maintenance = maintenanceForRestartType(rst)
+}
+
+func (r *respJSON) addWarningCount(count int, stamp time.Time) {
+	if r.WarningCount != 0 {
```

Comment:
Is this to prevent setting warning count twice? Should it error or panic instead?

Reply:
I just kept it because it was in the original code, in practice there is one place using this. Last wins would also work me.